src/inspector/network_agent.cc

```diff
@@ -329,10 +341,38 @@ protocol::DispatchResponse NetworkAgent::streamResourceContent(
     // If the request is finished, remove the entry.
     requests_.erase(in_requestId);
   }
-
   return protocol::DispatchResponse::Success();
 }
 
+protocol::DispatchResponse NetworkAgent::loadNetworkResource(
+    const protocol::String& in_url,
+    std::unique_ptr<protocol::Network::LoadNetworkResourcePageResult>*
+        out_resource) {
+  if (!env_->options()->experimental_inspector_network_resource) {
+    return protocol::DispatchResponse::ServerError(
+        "Network resource loading is not enabled. This feature is "
+        "experimental and requires --experimental-inspector-network-resource "
+        "flag to be set.");
+  }
+  std::string data = network_resource_manager_->Get(in_url);
```

Comment:
We should check that `network_resource_manager_` is not null here.

```suggestion
  CHECK_NOT_NULL(network_resource_manager_);
  std::string data = network_resource_manager_->Get(in_url);
```

Or, we can do an alternative check and get rid of `env_`:

```cpp
  if (network_resource_manager_ == nullptr) {
    return protocol::DispatchResponse::ServerError(
        "Network resource loading is not enabled. This feature is "
        "experimental and requires --experimental-inspector-network-resource "
        "flag to be set.");
  }
  std::string data = network_resource_manager_->Get(in_url);
```